tests/security/swtpm/swtpm_env_setup.pm

```diff
@@ -0,0 +1,106 @@
+#          OS images for later tests, prepare a script for remote
+#          ssh login and execution
+# Maintainer: rfan1 <richard.fan@suse.com>
+# Tags: poo#81256, tc#1768671
+
+use base 'opensusebasetest';
+use strict;
+use warnings;
+use testapi;
+use utils 'zypper_call';
+
+sub run {
+    my $self = shift;
+    $self->select_serial_terminal;
+
+    # Install qemu,libvirt,swtpm,expect and virsh_tools packages
+    zypper_call "in qemu libvirt swtpm expect virt-install virt-manager";
+    my $results = script_run("zypper --non-interactive in net-tools-deprecated");
+    if ($results) {
+        record_soft_failure("bsc#1180446 - can not install package 'net-tools-deprecated'");
+    }
+
+    # Start libvirtd daemon and start the default libvirt network
+    assert_script_run "systemctl start libvirtd";
+    assert_script_run "virsh net-start default";
+    assert_script_run "systemctl is-active libvirtd | grep active";
+    assert_script_run "virsh net-list | grep default | grep active";
+
+    # Download the pre-installed guest image with OS version SLE15SP3
+    my $img_url      = 'http://migration-smt2-1.qa.suse.de';
+    my $legacy_image = 'sle-15-SP3-x86_64-textmode_swtpm@64bit.qcow2';
+    my $uefi_image   = 'sle-15-SP3-x86_64-textmode_swtpm-uefi@64bit.qcow2';
+    my $legacy_xml   = 'swtpm_legacy.xml';
+    my $uefi_xml     = 'swtpm_uefi.xml';
+    my $image_path   = '/var/lib/libvirt/images';
+    foreach my $i ($legacy_image, $uefi_image, $legacy_xml, $uefi_xml) {
+        assert_script_run("wget -c -P $image_path $img_url/$i", 900);
+    }
+
+    # Write a expect script to implenet ssh access into remote host and run some commands"
+    my $ssh_script_1 = 'ssh_script_1';
+    my $ssh_script_2 = 'ssh_script_2';
+    assert_script_run(
+        'echo "$(cat <<EOF
+#!/usr/bin/expect
+
+#Usage sshsudologin.expect <host> <ssh user> <ssh password> <su user> <su password>
+
+set timeout 60
+spawn ssh [lindex \$argv 1]@[lindex \$argv 0]
+expect "yes/no*" {
+    send "yes\r"
+    expect "*?assword" { send "[lindex \$argv 2]\r" }
+    } "*?assword" { send "[lindex \$argv 2]\r" }
+
+expect " #" { send "ls /dev/tpm*\r" }
+expect " #" { send "tpm_version\r" }
+expect " #" { send "exit\r" }
+EOF
+        )" >' . "$image_path/$ssh_script_1"
+    );
+    assert_script_run(
+        'echo "$(cat <<EOF
+#!/usr/bin/expect
+
+#Usage sshsudologin.expect <host> <ssh user> <ssh password> <su user> <su password>
+
+set timeout 60
+spawn ssh [lindex \$argv 1]@[lindex \$argv 0]
+expect "yes/no*" {
+    send "yes\r"
+    expect "*?assword" { send "[lindex \$argv 2]\r" }
+    } "*?assword" { send "[lindex \$argv 2]\r" }
+
+expect " #" { send "ls /dev/tpm*\r" }
+expect " #" { send "tpm2_pcrread sha1:0\r" }
+expect " #" { send "exit\r" }
+EOF
+        )" >' . "$image_path/$ssh_script_2"
+    );
+    assert_script_run "chmod 755 $image_path/$ssh_script_1";
+    assert_script_run "chmod 755 $image_path/$ssh_script_2";
+}
```

Comment:
The purpose is these 2 scripts running in the remote site?  Actually, I do not really understand this part, would you please add more comments here :-)

Reply:
sure